### PR TITLE
Feature - NSLog and Print ConsoleWriter

### PR DESCRIPTION
**Pull request by cnoon**
This PR adds a `Method` enum to the `ConsoleWriter` class to allow you to select whether to use `print` or `NSLog` internally. The differences as to using one vs. the other is called out in the enum docstring.

The reason to switch between these is that you want to use either the `os_log` APIs or `NSLog` when running on a device. Otherwise, none of the log messages will appear in the device console.

### CODE REVIEW

File: Source/LogMessageWriter.swift [hunk 1]
```diff
@@ -37,10 +37,27 @@ public protocol LogMessageWriter {
 /// The ConsoleWriter class runs all modifiers in the order they were created and prints the resulting message
 /// to the console.
 open class ConsoleWriter: LogMessageWriter {
+    /// Used to define whether to use the print or NSLog functions when logging to the console.
+    ///
+    /// During development, it is recommended to use the `.print` case. When deploying to production, the `.nslog`
+    /// case should be used.
+    ///
+    /// - print: The Swift `print` function that is faster, not thread safe and doesn't write to the device console.
```
Comment (AtomicCat) [inline Source/LogMessageWriter.swift]: This would imply that if you have multiple loggers not all using the same queue, the default of `.print` may lead to crashes?

Comment (cnoon) [reply Source/LogMessageWriter.swift]: Whoops, let me rephrase. Good catch!

Comment (cnoon) [reply Source/LogMessageWriter.swift]: Fixed in 8f015d02.

**Comment by cnoon** [comment]
All comments have been addressed, I'm going to move this PR through. 

Thanks gents! 🍻
